client/hosting/sites/hooks/use-sync-selected-site.ts

```diff
@@ -10,9 +10,16 @@ export function useSyncSelectedSite(
 	selectedSite: SiteDetails | null | undefined
 ) {
 	const dispatch = useDispatch();
+	const isInitialLoad = useRef( true );
 
 	// Update selected site globally as soon as it is clicked from the table.
 	useEffect( () => {
+		// Prevent clearing the selected site on initial load.
+		if ( isInitialLoad.current && ! dataViewsState.selectedItem ) {
```

Comment:
I think we can just call `setSelectedSiteId` only when there is the selected item

Reply:
Removed the `isInitialLoad` condition for now, ensuring the masterbar always shows the last selected site.
https://github.com/Automattic/wp-calypso/pull/92720/commits/f93b7baf63eddceeca7ce9cb104164ab647a4cb5
p1721194766460809/1721157817.028909-slack-C06DN6QQVAQ

Reply:
📝 I reverted the above change: https://github.com/Automattic/wp-calypso/pull/92720#issuecomment-2232605959 and created the follow-up issue: https://github.com/Automattic/dotcom-forge/issues/8325

Reply:
> Is there any issue without the check of isInitialLoad?

If we are coming from outside the dashboard (like my home), without the initial load check this hook clears the selected site in claypso state.